src/main/UI/PasswordGeneratorBox.java

```diff
@@ -0,0 +1,134 @@
+        dialogBox.jframe.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        dialogBox.jframe.setResizable(false);
+        Color c = new Color(204,255,255);
+        dialogBox.jframe.getContentPane().setBackground(c);
+
+        // Create a label
+        label = new Label();
+        label.createLabel(90,40,150,20,dialogBox.jframe, "Password Length:", Color.blue);
+
+        //creating a Combobox to get the password length from the user
+        comboBox = new ComboBox();
+        String[] comboBoxItems = {"8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20"};
+        comboBox.createComboBox(dialogBox.jframe, comboBoxItems, Color.white,230,40,50,20);
+        comboBox.comboBox.setForeground(Color.black);
+
+        //Creating radio checkboxes to get the password type from the user
+        //Creating radiobox for Lowercase letters
+        lcbox = new RadioBox();
+        lcbox.createRadioButton(dialogBox.jframe, "Lowercase", Color.white,110,70,170,30);
+        lcbox.radioButton.setForeground(Color.black);
+        lcbox.radioButton.setBackground(c);
+        //Creating radiobox for Uppercase letters
+        ucbox = new RadioBox();
+        ucbox.createRadioButton(dialogBox.jframe, "Uppercase", Color.white,110,100,170,30);
+        ucbox.radioButton.setForeground(Color.black);
+        ucbox.radioButton.setBackground(c);
+        //Creating radiobox for numbox
+        numbox = new RadioBox();
+        numbox.createRadioButton(dialogBox.jframe, "Numbers", Color.white,110,130,150,30);
+        numbox.radioButton.setForeground(Color.black);
+        numbox.radioButton.setBackground(c);
+        //Creating radiobox for Special characters
+        spcbox = new RadioBox();
+        spcbox.createRadioButton(dialogBox.jframe, "Special Characters", Color.white,110,160,150,30);
+        spcbox.radioButton.setForeground(Color.black);
+        spcbox.radioButton.setBackground(c);
+
+        //Creating Button to generate the password
+        icon = new Icon();
+        icon.createIcon("../resources/icons/password.png");
+
+        button = new Button();
+        button.createButtonWithIcon(dialogBox.jframe, "Generate Password",icon.icon,130,195,100,30 );
+        button.button.addActionListener(this);
+        button.button.setForeground(Color.blue);
+        button.button.setBackground(Color.white);
+
+        //Creating a text Area to display the generated password
+        textArea = new TextArea();
+        textArea.createTextArea(dialogBox.jframe, "", Color.blue,40,240,310,35);
+        textArea.TextArea.setEditable(false);
+        textArea.TextArea.setAlignmentX(SwingConstants.CENTER);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        int passwordlen = Integer.parseInt(Objects.requireNonNull(comboBox.comboBox.getSelectedItem()).toString());
+        String choices = "";
+        if(spcbox.isSelected() && !lcbox.isSelected() && !ucbox.isSelected() && !numbox.isSelected()){
```

Comment:
It feels like there should be a better way! What would happen if we added a fifth option?

Reply:
Used String Concatenation to reform this part.